packages/extension/src/components/molecules/NetworkIndicator/NetworkIndicator.tsx

```diff
@@ -118,39 +201,132 @@ export const NetworkIndicator: FC = () => {
+              </DialogContentText>
+            </DialogContent>
+            <DialogActions>
+              <Button onClick={() => setConfirmDeleteOpen(false)} color="primary">
+                Cancel
+              </Button>
+              <Button onClick={handleConfirmDelete} color="primary" autoFocus>
+                Delete
+              </Button>
+            </DialogActions>
+          </Dialog>
+          <Dialog
+            open={activeNetworkDeleteDialogOpen}
+            onClose={() => setActiveNetworkDeleteDialogOpen(false)}
+          >
+            <DialogTitle>Error</DialogTitle>
+            <DialogContent>
+              <DialogContentText>
+                You are currently connected to the network {currentNetworkName}. Please switch to
+                another network before deleting this one.
+              </DialogContentText>
+            </DialogContent>
+            <DialogActions>
+              <Button
+                onClick={() => setActiveNetworkDeleteDialogOpen(false)}
+                color="primary"
+                autoFocus
+              >
+                OK
+              </Button>
+            </DialogActions>
+          </Dialog>
+          <Dialog
+            fullScreen
+            open={explanationOpen}
+            onClose={handleClose}
+            TransitionComponent={Transition}
+            data-testid="network-indicator-dialog"
+          >
+            <AppBar sx={{ position: 'relative' }}>
+              <Toolbar>
+                <IconButton edge="start" color="inherit" onClick={handleClose} aria-label="close">
+                  <CloseIcon />
+                </IconButton>
+                <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
+                  Change Network
+                </Typography>
+              </Toolbar>
+            </AppBar>
+            <div style={{ overflowY: 'scroll', height: '544px', margin: '20px 20px 0 20px' }}>
+              <div style={{ paddingBottom: '40px' }}>
+                {
+                  // Display all the pre-defined networks
+                  Object.keys(NETWORK)
+                    .filter((network) => network !== Network.CUSTOM)
+                    .map((_network) => {
+                      const { name, server, description } = NETWORK[_network as Network];
+                      return (
+                        <NetworkDisplay
+                          key={_network}
+                          name={name}
+                          server={server}
+                          description={description}
+                          isSelected={name === currentNetworkName}
+                          onClick={() => handleClickOnNetwork(_network as Network)}
+                        />
+                      );
+                    })
+                }
+                {
+                  // Display the custom networks of the user
+                  Object.keys(existingNetworks).map((_network) => {
```

Comment:
I would extract this list of networks to a `useMemo` in order to don't compute this at each render.

Reply:
Done